packages/cloud/src/libraries/tenants.ts

```diff
@@ -67,6 +67,45 @@ export class TenantsLibrary {
     });
   }
 
+  async deleteTenantById(tenantId: string) {
+    const {
+      deleteTenantById,
+      deleteDatabaseRoleForTenant,
+      deleteResourcesByTenantId,
+      deleteRoleByTenantId,
+      getTenantById,
+      deleteM2MApplicationByTenantId,
+      removeUrisFromAdminConsoleRedirectUris,
+    } = this.queries.tenants;
+    const { cloudUrlSet } = EnvSet.global;
+
+    // `dbUser` is defined as nullable but we always specified this value when creating a new tenant.
+    const { dbUser } = await getTenantById(tenantId);
+    if (dbUser) {
+      await deleteDatabaseRoleForTenant(dbUser); // DB role for building connection for the current tenant.
+    }
+
+    await deleteTenantById(tenantId);
+
+    /**
+     * All applications, resources, scopes and roles attached to the current tenant
+     * will be deleted per DB design.
+     * Need to manually delete following applications, roles, resources since they
+     * are created for admin tenant which will not be deleted automatically.
+     */
+    // Delete management API for the current tenant.
+    // `scopes` will be automatically deleted if its related resources have been removed.
+    await deleteResourcesByTenantId(tenantId);
+    // Delete admin tenant admin role for the current tenant.
+    await deleteRoleByTenantId(tenantId);
+    // Delete M2M application for the current tenant.
+    await deleteM2MApplicationByTenantId(tenantId);
```

Comment:
Why only M2M application? Shouldn't we delete all apps from the given tenant?

Reply:
other applications will be created for the current tenant while m2m was created for admin tenant. Other applications will be automatically removed upon the removal of the current tenant